test/jasmine/tests/histogram_test.js

```diff
@@ -156,7 +156,7 @@ describe('Test histogram', function() {
                 nbinsx: 4
             });
 
-            // Note: this gives half-day gaps between all but the first two
+            // TODO: this gives half-day gaps between all but the first two
             // bars. Now that we have explicit per-bar positioning, perhaps
             // we should fill the space, rather than insisting on equal-width
             // bars?
```

Comment:
This effect is a little more important than getting the exact right bin sizes in the 2D case, because here it leads to visible gaps between many of the bars. Unfortunately it's also quite a bit more effort to solve when you consider stacked or grouped traces. Worth making an issue for it?